### PR TITLE
fix: sqlite `filename` config value was ignored

### PR DESCRIPTION
**Pull request by jaylmiller**
Screen Shot 2024-03-01 at 11 27 47 AM
Screen Shot 2024-03-01 at 11 27 57 AM

### CODE REVIEW

**Comment by 0xOlias** [comment]
Nice catch. Taking a look, I realize that we broke this a little while back when we separated the indexing store and sync store into separate database files (before it was just one filename). So now, the `filename` option is kind of a misnomer; there's no way for it to behave as expected.

IMO the fix is to rename `filename` to `directory`, but that's technically a breaking change. So we're going to include it in `0.3` (tagged you in a comment on that PR).

TL;DR you're totally right but I don't think it makes sense to fix in this way. Feel free to push back!

File: packages/core/src/config/database.ts [hunk 1]
```diff
@@ -28,10 +28,15 @@ export const buildDatabase = ({
...
-  const defaultStorePath = path.join(ponderDir, "store");
-  const defaultSyncFilePath = path.join(defaultStorePath, "sync.db");
-  const defaultIndexingFilePath = path.join(defaultStorePath, "indexing.db");
-  const sqlitePrintPath = path.relative(rootDir, defaultStorePath);
+  const storePath = (() => {
+    if (config.database?.kind === "sqlite" && config.database.filename) {
+      return path.join(rootDir, config.database.filename);
+    }
+    return path.join(ponderDir, "store");
+  })();
```
Comment (0xOlias) [inline packages/core/src/config/database.ts:36]: IMO a ternary here is probably more idiomatic for TS:
```ts
const storePath = config.database?.kind === "sqlite" && config.database.filename
  ? path.join(rootDir, config.database.filename);
  : path.join(ponderDir, "store");
```

**Review by 0xOlias — COMMENTED** [review]
Looks good! Only thing missing is a changeset.

File: packages/core/src/config/database.ts (hunk 1 above)
Comment (jaylmiller) [reply packages/core/src/config/database.ts:36]: good point

**Review by 0xOlias — APPROVED** [review]
LGTM, gz on first contribution!
